JavaScript development/Frameworks/React.md

```diff
@@ -0,0 +1,389 @@
+```
+
+### Exports
+- each JSX file should have a single exported component
+ 
+``` javascript
+export default class ComponentName extends Component {
+  // ...
+}
+    
+// or
+
+class ComponentName extends Component {
+  // ...
+}
+    
+export default ComponentName;
+```
+
+### Props Passing
+
+- use camelCase for props
+- use double quotes for JSX attribute values if you are passing them a *string*
+- use self-closing tags if the element doesn't have children
+
+``` javascript
+class ComponentName extends Component {
+    // ...
+    render () {
+        return (
+          <ChildComponent
+            defaultTitle="Some Title"
+            title={this.props.title}
+          />
+        );
+    }
+}
+```
+
+- declare functions before passing them as props instead of using anonymous functions:
+
+``` javascript
+// bad
+<ChildComponent onClick={(event) => {console.log(event)}} />
+    
+// good
+class ComponentName extends Component {
+    // ...
+    onClickHandler(event) {
+        console.log(event);
+    }
+    // ...
+    render () {
+        return (
+          <ChildComponent onClick={this.onClickHandler} />
+        );
+    }
+}
+```
+
+Omit boolean values if default value is **true**:
+
+``` javascript
+// bad
+<ChildComponent
+    onClick={this.onClickHandler}
+    isPropActive={true}
+/>
+
+// good
+<ChildComponent
+    onClick={this.onClickHandler}
+    isPropActive
+/>
+```
+
+### Props Receiving
+- always use type checking for props
+- use `defaultProps` if you need to set default values in cases when the prop value isn't received
+
+``` javascript
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+class ComponentName extends Component {
+    // ...
+    render() {
+        return (
+          <div>{this.props.name}</div>
+        );
+      }
+}
+
+ComponentName.propTypes = {
+    id: PropTypes.string.isRequired,
+    name: PropTypes.string,
+};
+
+ComponentName.defaultProps = {
+    name: 'Default',
+};
+```
+
+- you can also declare property types as a **static member** inside a class:
+
+``` javascript
+class ComponentName extends Component {
+    static propTypes = {
+        id: PropTypes.string.isRequired,
+        name: PropTypes.string,
+    }
+    // ...
+    render() {
+        return (
+          <div>{this.props.name}</div>
+        );
+      }
+}
+```
+
+- inside the *render* method, preffer accessing the props using **object destructuring**:
+
+``` javascript
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+class ComponentName extends Component {
+    // ...
+    render() {
+        const {id, name} = this.props;
+        // ...
+        return (
+          <div>{name}</div>
+        );
+    }
+}
+```
+
+### Component Local State
+- if you have local state modified by the component, preffer using a single object member
+
+``` javascript
+class ComponentName extends Component {
+    // ...
+    
+    const componentState = {
```

Comment:
This example doesn't make sense:

- If you're using pure React, you would use `setState` and in this case, you would need to use the `state` property
- This is ok for MobX local state, but you should then mention this